src/components/RequestsFilters/RequestsFilters.test.js

```diff
@@ -0,0 +1,204 @@
+import {
+  render,
+  screen,
+} from '@folio/jest-config-stripes/testing-library/react';
+import userEvent from '@folio/jest-config-stripes/testing-library/user-event';
+
+import {
+  Accordion,
+} from '@folio/stripes/components';
+import {
+  CheckboxFilter,
+  MultiSelectionFilter,
+} from '@folio/stripes/smart-components';
+
+import RequestsFilters from './RequestsFilters';
+import { RequestLevelFilter } from './RequestLevelFilter';
+import { PickupServicePointFilter } from './PickupServicePointFilter';
+
+import {
+  requestFilterTypes,
+} from '../../constants';
+
+jest.mock('./RequestLevelFilter', () => ({
+  RequestLevelFilter: jest.fn((props) => (<div {...props} />)),
+}));
+jest.mock('./PickupServicePointFilter', () => ({
+  PickupServicePointFilter: jest.fn((props) => (<div {...props} />)),
+}));
+jest.mock('@folio/stripes/smart-components', () => ({
+  CheckboxFilter: jest.fn((props) => (<div {...props} />)),
+  MultiSelectionFilter: jest.fn((props) => (<div {...props} />)),
+}));
+
+const onChange = jest.fn();
+const onClear = jest.fn();
+const props = {
+  activeFilters: {
+    requestStatus: ['Open'],
+    requestType: ['Hold'],
+    pickupServicePoints: ['1'],
+    tags: ['Urgent'],
+    requestLevels: [],
+  },
+  onChange,
+  onClear,
+  resources: {
+    servicePoints: {
+      records: [{
+        id: '1',
+        name: 'Service Point 1',
+      }],
+    },
+    tags: {
+      records: [{
+        label: 'Urgent',
+      }],
+    },
+  },
+  titleLevelRequestsFeatureEnabled: false,
+};
+const accordionSelector = [
+  requestFilterTypes.REQUEST_TYPE,
+  requestFilterTypes.REQUEST_STATUS,
+  requestFilterTypes.TAGS,
+];
+const testIds = {
+  [requestFilterTypes.REQUEST_TYPE]: requestFilterTypes.REQUEST_TYPE,
+  [`${requestFilterTypes.REQUEST_TYPE}Filter`]: `${requestFilterTypes.REQUEST_TYPE}Filter`,
+  [requestFilterTypes.REQUEST_STATUS]: requestFilterTypes.REQUEST_STATUS,
+  [`${requestFilterTypes.REQUEST_STATUS}Filter`]: `${requestFilterTypes.REQUEST_STATUS}Filter`,
+  [requestFilterTypes.TAGS]: requestFilterTypes.TAGS,
+  requestLevelFilter: 'requestLevelFilter',
+  multiSelectionFilter: 'multiSelectionFilter',
+  pickupServicePointFilter: 'pickupServicePointFilter',
+};
+const labelIds = {
+  [requestFilterTypes.REQUEST_TYPE]: 'ui-requests.requestMeta.type',
+  [requestFilterTypes.REQUEST_STATUS]: 'ui-requests.requestMeta.status',
+  [requestFilterTypes.TAGS]: 'ui-requests.requestMeta.tags',
+};
+
+describe('RequestsFilters', () => {
+  beforeEach(() => {
+    render(<RequestsFilters {...props} />);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  accordionSelector.forEach((requestFilterType, index) => {
```

Comment:
I think over-engineering tests makes it harder to undestand what is being tested, and tests need to be readable and clearly show what part of tested code fails. With such logic a developer first needs to figure out how the tests are implemented and it's going to take away time that could be used for fixing actual bugs.

I understand how it's tempting to make everything DRY, but it doesn't apply to tests in the same way as to business logic. And for just three accordions it's not really that much repeated test code:

```
describe('Request type accordion', () => {
  it(`should trigger Request type accordion with correct props`, () => {
    expect(Accordion).toHaveBeenNthCalledWith(1, expect.objectContaining({
      displayClearButton: true,
      label: labelIds[requestFilterTypes.REQUEST_TYPE],
      id: requestFilterTypes.REQUEST_TYPE,
      'data-testid': requestFilterTypes.REQUEST_TYPE,
      name: requestFilterTypes.REQUEST_TYPE,
      onClearFilter: expect.any(Function),
      header: expect.any(Function),
      separator: false,
    }), {});
  });

  it(`should be called onClearFilter with request filter types Request type`, async () => {
    await userEvent.click(screen.getByTestId(`${testIds[requestFilterTypes.REQUEST_TYPE]}Button`));

    expect(onClear).toHaveBeenCalledWith(requestFilterTypes.REQUEST_TYPE);
  });

  it(`should render CheckboxFilter for Request type accordion`, () => {
    expect(screen.getByTestId(testIds[`${requestFilterTypes.REQUEST_TYPE}Filter`])).toBeInTheDocument();
  });

  it(`should trigger CheckboxFilter for Request type accordion with correct props`, () => {
    expect(CheckboxFilter).toHaveBeenNthCalledWith(1, expect.objectContaining({
      'data-testid': testIds[`${requestFilterTypes.REQUEST_TYPE}Filter`],
      name: requestFilterTypes.REQUEST_TYPE,
      selectedValues: props.activeFilters[requestFilterTypes.REQUEST_TYPE],
      onChange,
    }), {});
  });
});

describe('Request status accordion', () => {
  it(`should trigger Request status accordion with correct props`, () => {
    expect(Accordion).toHaveBeenNthCalledWith(2, expect.objectContaining({
      displayClearButton: true,
      label: labelIds[requestFilterTypes.REQUEST_STATUS],
      id: requestFilterTypes.REQUEST_STATUS,
      'data-testid': requestFilterTypes.REQUEST_STATUS,
      name: requestFilterTypes.REQUEST_STATUS,
      onClearFilter: expect.any(Function),
      header: expect.any(Function),
      separator: false,
    }), {});
  });

  it(`should be called onClearFilter with request filter types Request status`, async () => {
    await userEvent.click(screen.getByTestId(`${testIds[requestFilterTypes.REQUEST_STATUS]}Button`));

    expect(onClear).toHaveBeenCalledWith(requestFilterTypes.REQUEST_STATUS);
  });

  it(`should render CheckboxFilter for Request status accordion`, () => {
    expect(screen.getByTestId(testIds[`${requestFilterTypes.REQUEST_STATUS}Filter`])).toBeInTheDocument();
  });

  it(`should trigger CheckboxFilter for Request status accordion with correct props`, () => {
    expect(CheckboxFilter).toHaveBeenNthCalledWith(2, expect.objectContaining({
      'data-testid': testIds[`${requestFilterTypes.REQUEST_STATUS}Filter`],
      name: requestFilterTypes.REQUEST_STATUS,
      selectedValues: props.activeFilters[requestFilterTypes.REQUEST_STATUS],
      onChange,
    }), {});
  });
});

describe('Tags accordion', () => {
  it(`should trigger Tags accordion with correct props`, () => {
    expect(Accordion).toHaveBeenNthCalledWith(3, expect.objectContaining({
      displayClearButton: true,
      label: labelIds[requestFilterTypes.TAGS],
      id: requestFilterTypes.TAGS,
      'data-testid': requestFilterTypes.TAGS,
      name: requestFilterTypes.TAGS,
      onClearFilter: expect.any(Function),
      header: expect.any(Function),
      separator: false,
    }), {});
  });

  it(`should be called onClearFilter with request filter types Tags`, async () => {
    await userEvent.click(screen.getByTestId(`${testIds[requestFilterTypes.TAGS]}Button`));

    expect(onClear).toHaveBeenCalledWith(requestFilterTypes.TAGS);
  });

  it(`should render MultiSelectionFilter for Tags accordion`, () => {
    expect(screen.getByTestId(testIds.multiSelectionFilter)).toBeInTheDocument();
  });

  it(`should trigger MultiSelectionFilter for Tags accordion with correct props`, () => {
    expect(MultiSelectionFilter).toHaveBeenCalledWith(expect.objectContaining({
      'data-testid': testIds.multiSelectionFilter,
      dataOptions: [{
        label: 'Urgent',
        value: 'Urgent',
      }],
      name: requestFilterTypes.TAGS,
      selectedValues: props.activeFilters.tags,
      onChange,
      ariaLabelledBy: requestFilterTypes.TAGS,
    }), {});
  });
});
```

It's only about 30 lines more, but so much more easier to understand what each test is for

Reply:
Fixed